### PR TITLE
feat: hypixel speed update w/ strafe

### PR DESCRIPTION
**Pull request by liquidsquid1**
- made the horizontal and vertical acceleration customisable
- added a strafe bypass, this should be stable enough on 0.7 but lowering it may help reduce random flags

### CODE REVIEW

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/speed/modes/watchdog/SpeedHypixelBHop.kt [hunk 1]
```diff
@@ -46,15 +46,9 @@ object SpeedHypixelBHop : Choice("HypixelBHop") {
...
-    private val verticalAcceleration by boolean("VerticalAcceleration", true)
-
-    private const val BASE_HORIZONTAL_MODIFIER = 0.0004
-
-    // todo: check if we can do more with this
-    private const val HORIZONTAL_SPEED_AMPLIFIER = 0.0007
-    private const val VERTICAL_SPEED_AMPLIFIER = 0.0004
-
+    private val horizontalAcceleration by float("HorizontalAcceleration", 0.00F, 0.00F..1.00F)
+    private val verticalAcceleration by float("VerticalAcceleration", 0.01F, 0.00F..1.00F)
```
Comment (Katinuka) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/speed/modes/watchdog/SpeedHypixelBHop.kt]: There is a problem. It's impossible to set such precise values (like 0.0007 or  0.0004) with those values represented as sliders in the GUI. Users will probably be in a situation where 0.00 is too little and 0.01 is too much.

Comment (liquidsquid1) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/speed/modes/watchdog/SpeedHypixelBHop.kt]: I have thought about that..

The best course of action would probably be to make an initial value for the horizontal (and vertical) speed modifier that can simply be added onto with the slider, but that would render this speed useless for every other server apart from hypixel due to the vertical modification

Comment (liquidsquid1) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/speed/modes/watchdog/SpeedHypixelBHop.kt]: Or I could just increase the scale of the horizontal acceleration and then divide it when it’s used

Comment (liquidsquid1) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/speed/modes/watchdog/SpeedHypixelBHop.kt]: I will resolve this by simply making the slider with lower bounds and by multiplying the initial horizontal acceleration

**Comment by liquidsquid1** [comment]
marked as a draft as apparently the strafe flags for people 🗿

**Comment by EclipsesDev** [comment]
well yeah, 0.7 strafe is to much is it? also did u even test this in hypixel 🗿

**Comment by liquidsquid1** [comment]
> well yeah, 0.7 strafe is to much is it? also did u even test this in hypixel 🗿

0.7 on falling works quite reliably on newer versions.

**Comment by liquidsquid1** [comment]
# seems like this strafe has been patched by hypixel


i will find another bypass, leaving this as a draft in the meantime.

**Comment by 1zun4** [comment]
This PR likely has been replaced by https://github.com/CCBlueX/LiquidBounce/pull/3380 - make a new PR if you got something new.

**Pull request closed without merging**